packages/atomic/src/components/common/items-per-page/validate.ts

```diff
@@ -0,0 +1,32 @@
+import {AtomicCommerceProductsPerPage} from '../../commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page';
+import {AtomicResultsPerPage} from '../../search/atomic-results-per-page/atomic-results-per-page';
+
+export function validateChoicesDisplayed(
+  context: AtomicResultsPerPage | AtomicCommerceProductsPerPage
+) {
+  return context.choicesDisplayed.split(',').map((choice) => {
+    const parsedChoice = parseInt(choice);
+    if (isNaN(parsedChoice)) {
+      const errorMsg = `The choice value "${choice}" from the "choicesDisplayed" option is not a number.`;
+      context.bindings.engine.logger.error(errorMsg, context);
+      throw new Error(errorMsg);
+    }
+
+    return parsedChoice;
+  });
+}
+
+export function validateInitialChoice(
```

Comment:
I'd call this function "getInitialPageSize" instead.

Reply:
instead I removed the side effect !